airbyte-cdk/python/airbyte_cdk/sources/declarative/retrievers/simple_retriever.py

```diff
@@ -325,10 +326,13 @@ def read_records(
             if self.cursor and current_record:
                 self.cursor.observe(_slice, current_record)
 
+            # Latest record read, not necessarily within slice boundaries.
+            # TODO Remove once all custom components implement `observe` method.
```

Comment:
@lazebnyi can you create a follow up issue so we can take care of this?

Reply:
created - https://github.com/airbytehq/airbyte-internal-issues/issues/6955